controllers/storagecluster/reconcile.go

```diff
@@ -433,6 +433,15 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			message := fmt.Sprintf("Error while reconciling: %v", returnErr)
 			statusutil.SetErrorCondition(&instance.Status.Conditions, reason, message)
 
+			// if the error was due to skipped storage classes, we shouldn't set the phase to error
+			// instead of error phase we will set it to initializing
+			if strings.Contains(returnErr.Error(), "some StorageClasses were skipped while waiting for pre-requisites to be met") {
+				if instance.Status.Phase != statusutil.PhaseOnboarding {
+					instance.Status.Phase = statusutil.PhaseInitializing
```

Comment:
Instead of PhaseInitializing can not we make use of progressing itself

Reply:
We can actually, I wanted to differentiate between these two phases actually. Basically, the Initializing phase suggests the storageclass creation is happening. That's why thought of the new phase. If you think the progressing phase will be suitable for this, then I can use that instead.

Reply:
Earlier we were seeing the progressing phase only when this bug was not present. So I would suggest having the progressing phase.

Reply:
Have Updated the PR, Please Take anotherLook